packages/wallet/backend/src/rafiki/auth/service.ts

```diff
@@ -0,0 +1,62 @@
+import { Logger } from 'winston'
+import { Env } from '@/config/env'
+import { GraphQLClient } from 'graphql-request'
+import {
+  getGrantByIdQuery,
+  getGrantsQuery,
+  revokeGrantMutation
+} from '@/rafiki/auth/request/grant.request'
+import {
+  GetGrantQuery,
+  GetGrantQueryVariables,
+  GetGrantsQuery,
+  GetGrantsQueryVariables,
+  Grant,
+  RevokeGrantMutation,
+  RevokeGrantMutationVariables
+} from '@/rafiki/auth/generated/graphql'
+
+interface IRafikiAuthService {
+  listGrants(): Promise<Grant[]>
+  getGrantById(id: string): Promise<Grant>
+  revokeGrant(id: string): Promise<void>
+}
+
+interface RafikiAuthServiceDependencies {
+  logger: Logger
+  env: Env
+  gqlClient: GraphQLClient
+}
+
+export class RafikiAuthService implements IRafikiAuthService {
+  constructor(private deps: RafikiAuthServiceDependencies) {}
+
+  async listGrants() {
+    const response = await this.deps.gqlClient.request<
+      GetGrantsQuery,
+      GetGrantsQueryVariables
+    >(getGrantsQuery, {})
+
+    return response.grants.edges.map((el: { node: Grant }) => el.node)
```

Comment:
is edges always returned as an empty list? because if it is also undefined, this can break with map of undefined